openc3-init/plugins/packages/openc3-tool-cmdsender/src/tools/CommandSender/CommandSender.vue

```diff
@@ -471,56 +470,61 @@ export default {
           } else if (this.isArray(str)) {
             return eval(str)
           } else {
-            return str
+            return `'${str}'`
           }
         }
       } else {
-        return quotes_removed
+        return `'${quotesRemoved}'`
       }
     },
 
     convertToString(value) {
       var i = 0
-      var return_value = ''
+      var returnValue = ''
       if (Object.prototype.toString.call(value).slice(8, -1) === 'Array') {
         var arrayLength = value.length
-        return_value = '[ '
+        returnValue = '[ '
         for (i = 0; i < arrayLength; i++) {
           if (
             Object.prototype.toString.call(value[i]).slice(8, -1) === 'String'
           ) {
-            return_value += '"' + value[i] + '"'
+            returnValue += '"' + value[i] + '"'
           } else {
-            return_value += value[i]
+            returnValue += value[i]
           }
           if (i != arrayLength - 1) {
-            return_value += ', '
+            returnValue += ', '
           }
         }
-        return_value += ' ]'
+        returnValue += ' ]'
       } else if (
         Object.prototype.toString.call(value).slice(8, -1) === 'Object'
       ) {
         if (value.json_class == 'String' && value.raw) {
           // This is binary data, display in hex.
-          return_value = '0x'
+          returnValue = '0x'
           for (i = 0; i < value.raw.length; i++) {
             var nibble = value.raw[i].toString(16).toUpperCase()
             if (nibble.length < 2) {
               nibble = '0' + nibble
             }
-            return_value += nibble
+            returnValue += nibble
           }
         } else if (value.json_class == 'Float' && value.raw) {
-          return_value = value.raw
+          returnValue = value.raw
         } else {
           // TBD - are there other objects that we need to handle?
-          return_value = String(value)
+          returnValue = String(value)
         }
       } else {
-        return_value = String(value)
+        returnValue = String(value)
+        // Ensure if this is a string that it is in quotes
+        // This is the case coming from the GUI but not from the cmd history
+        if (typeof value === 'string' && value.charAt(0) !== "'") {
+          returnValue = `'${returnValue}'`
+        }
```

Comment:
Note this also quotes our states like `cmd("INST COLLECT with TYPE 'NORMAL', DURATION 1, OPCODE 171, TEMP 0")` but it was the only way to make ASCII commands like `cmd("INST ASCIICMD with STRING 'BLAH BLAH'")` work. Good news is if you manually delete the quotes it still works: `cmd("INST COLLECT with TYPE NORMAL, DURATION 1, OPCODE 171, TEMP 0")`